src/storage/src/source/mod.rs

```diff
@@ -928,7 +941,16 @@ where
 
             bytes_read_counter.inc_by(bytes_read as u64);
             source_metrics.record_partition_offsets(metric_updates);
-            source_metrics.capability.set(*cap.time());
+            // WIP: Not sure taking the first of the contained caps is correct.
```

Comment:
We can put some logic here that is akin to Antichain's `into_option` and `as_option` which asserts that len <= 1.